components/table/table-col-sort-button.js

```diff
@@ -122,8 +141,19 @@ export class TableColSortButton extends FocusMixin(LitElement) {
 		const iconView = !this.nosort ?
 			html`<d2l-icon icon="${this.desc ? 'tier1:arrow-toggle-down' : 'tier1:arrow-toggle-up'}"></d2l-icon>` :
 			null;
+		const buttonTitle = this._getSortButtonTitle();
+		const description = TableColSortButton.#buttonDescription[Number(this.nosort)];
+
+		return html`<button aria-description="${description}" aria-label="${buttonTitle}" title="${buttonTitle}" type="button"><slot></slot>${iconView}</button>`;
+	}
+
+	_getSortButtonTitle() {
+		if (!this.nosort) {
+			const sortKey = TableColSortButton.#columnDataTypeKeys[this.columnDataType][Number(!this.desc)];
+			return this.localize(sortKey);
+		}
 
-		return html`<button type="button"><slot></slot>${iconView}</button>`;
+		return '';
```

Comment:
With how this is currently done, I think it would still render with an empty `aria-label`. What we'd generally want to do is set the value to `undefined` in the case where there wouldn't be a label. Then in the html we'd use `aria-label="${ifDefined(buttonTitle)}"`.

Also with this function, you could early exit with an `if (this.nosort) return undefined` and then you don't have to indent the rest.